### PR TITLE
Mkheranidev1

### PR DESCRIPTION
**Pull request by mkherani**
PR for Get-RecoveryPoint.

Pending Items:
1. Testing.
2. String msg (help msg, logs, exception message)

### CODE REVIEW

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                    parameter.Add(GetRecoveryPointParams.EndDate, EndDate);
+                    PsBackupProviderManager providerManager = new PsBackupProviderManager(parameter, HydraAdapter);
+                    IPsBackupProvider psBackupProvider = providerManager.GetProviderInstance(Item.ContainerType, Item.BackupManagementType);
+                    var rpList = psBackupProvider.ListRecoveryPoints();
+                    if (rpList.Count == 1)
+                        WriteObject(rpList[0]);
+                    else
+                        WriteObject(rpList);
+                }
+
```
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs]: Nit: Extra new line.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                        throw new Exception("End date should be greated than start date"); //tbd: Correct nsg and exception type
+                    }
+
+                    parameter.Add(GetRecoveryPointParams.StartDate, StartDate);
+                    parameter.Add(GetRecoveryPointParams.EndDate, EndDate);
+                    PsBackupProviderManager providerManager = new PsBackupProviderManager(parameter, HydraAdapter);
+                    IPsBackupProvider psBackupProvider = providerManager.GetProviderInstance(Item.ContainerType, Item.BackupManagementType);
+                    var rpList = psBackupProvider.ListRecoveryPoints();
+                    if (rpList.Count == 1)
+                        WriteObject(rpList[0]);
```
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs]: Nit: Please enclose in curly braces.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+                Dictionary<System.Enum, object> parameter = new Dictionary<System.Enum, object>();
+                parameter.Add(GetRecoveryPointParams.Item, Item);
+
+                if(this.ParameterSetName == DateTimeFilterParameterSet)
+                {
+                    //User want list of RPs between given time range
+                    if (StartDate >= EndDate)
+                    {
+                        throw new Exception("End date should be greated than start date"); //tbd: Correct nsg and exception type
```
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs]: I added classes for different types of messages we have to print. Please add this as a variable in the same. It'll be easier later to give these to PM and get them scrubbed.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRMRecoveryServicesRecoveryPoint"), OutputType(typeof(List<AzureRmRecoveryServicesRecoveryPointBase>), typeof(AzureRmRecoveryServicesRecoveryPointBase))]
+    public class GetAzureRMRecoveryServicesRecoveryPoint : RecoveryServicesBackupCmdletBase
+    {
+        internal const string DateTimeFilterParameterSet = "DateTimeFilter";
+        internal const string RecoveryPointIdParameterSet = "RecoveryPointId";
+
+        [Parameter(Mandatory = true, ParameterSetName = DateTimeFilterParameterSet, HelpMessage = "", ValueFromPipeline = false)]
```
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs]: When you add, please add help messages in the common class.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs (hunk 3 above)
Comment (mkherani) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs]: Sure. All msg are pending i just added some msg. I will go update the msg in class during fit and finish (after testing).

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs (hunk 4 above)
Comment (mkherani) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/RecoveryPoint/GetAzureRMRecoveryServicesRecoveryPoint.cs]: sure.

**Pull request merged**